### PR TITLE
Create new users 164513784

### PR DESCRIPTION
**Pull request by justgarcia5**

### CODE REVIEW

File: app/assets/javascripts/users.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (jasononaquest) [inline app/assets/javascripts/users.coffee]: please remove empty files

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
```
Comment (jasononaquest) [inline app/assets/stylesheets/users.scss]: please remove empty files

File: app/assets/stylesheets/scaffolds.scss [hunk 3]
```diff
@@ -0,0 +1,84 @@
+body {
```
Comment (jasononaquest) [inline app/assets/stylesheets/scaffolds.scss]: Please verify that this file is being used. If it isn't, you can remove it.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+
+  # GET /users
```
Comment (jasononaquest) [inline app/controllers/users_controller.rb]: Although not necessary, removing this comments is generally acceptable.

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (jasononaquest) [inline app/helpers/users_helper.rb]: please remove empty files

File: app/views/users/_user.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1,2 @@
+json.extract! user, :id, :created_at, :updated_at
```
Comment (jasononaquest) [inline app/views/users/_user.json.jbuilder]: What is this file used for?

File: app/views/users/index.json.jbuilder [hunk 7]
```diff
@@ -0,0 +1 @@
+json.array! @users, partial: 'users/user', as: :user
```
Comment (jasononaquest) [inline app/views/users/index.json.jbuilder]: What is this file used for?

File: app/views/users/show.json.jbuilder [hunk 8]
```diff
@@ -0,0 +1 @@
+json.partial! "users/user", user: @user
```
Comment (jasononaquest) [inline app/views/users/show.json.jbuilder]: What is this file used for?

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  def show
+  end
+
+  # GET /users/new
+  def new
+    @user = User.new
+  end
+
+  # GET /users/1/edit
+  def edit
```
Comment (jasononaquest) [inline app/controllers/users_controller.rb]: We only want to commit code that has functionality at this point in the application (we can't edit, update, or destroy yet).

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    end
+  end
+
+  # PATCH/PUT /users/1
+  # PATCH/PUT /users/1.json
+  def update
+    respond_to do |format|
+      if @user.update(user_params)
+        format.html { redirect_to @user, notice: 'User was successfully updated.' }
+        format.json { render :show, status: :ok, location: @user }
```
Comment (jasononaquest) [inline app/controllers/users_controller.rb]: We're not using JSON responses in this app.

File: app/views/users/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<p id="notice"><%= notice %></p>
+
+<%= link_to 'Edit', edit_user_path(@user) %> |
```
Comment (jasononaquest) [inline app/views/users/show.html.erb]: We aren't updating users yet

File: spec/features/user_spec.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+require 'rails_helper'
+require 'capybara/rspec'
+
+feature 'Admin creates a new user' do
+  scenario 'with fullname, email, is_admin, team_leader_id' do
+    visit new_user_path
+    fill_in 'Full Name', with: 'Test'
+    fill_in 'Email', with: 'email@email.com'
+
```
Comment (jasononaquest) [inline spec/features/user_spec.rb]: We want to make sure our form captures all the attributes from the story acceptance criteria

File: app/assets/stylesheets/zeal-initiative-tracker.webflow.css [hunk 13]
```diff
@@ -25,6 +25,7 @@ h3 {
   margin-bottom: 0px;
   font-size: 24px;
   font-weight: 700;
+
```
Comment (jasononaquest) [inline app/assets/stylesheets/zeal-initiative-tracker.webflow.css]: Please remove empty lines within style block

File: app/assets/stylesheets/zeal-initiative-tracker.webflow.css [hunk 14]
```diff
@@ -288,6 +291,9 @@ h5 {
   font-family: Lato, sans-serif;
   font-size: 16px;
   line-height: 1.4;
+  position: relative;
+  z-index: 1;
+
```
Comment (jasononaquest) [inline app/assets/stylesheets/zeal-initiative-tracker.webflow.css]: Please remove empty lines within style block

File: app/views/users/edit.html.erb [hunk 15]
```diff
@@ -1,6 +1,6 @@
-# <h1>Editing User</h1>
+<h1>Editing User</h1>
```
Comment (jasononaquest) [inline app/views/users/edit.html.erb]: It looks like we removed the #edit method in the controller so is this file still needed?

File: app/views/users/index.html.erb [hunk 16]
```diff
@@ -8,6 +8,7 @@
   <div class="init-list">
   <% @users.each do |user| %>
     <div class="card">
+      <%= link_to edit_user_path(user), :style=> "text-decoration: none;" do %>
```
Comment (jasononaquest) [inline app/views/users/index.html.erb]: Do we need this edit link?

File: app/views/users/index.html.erb [hunk 17]
```diff
@@ -19,5 +20,7 @@
         <p class="init-team-leader">Team Member</p>
       <% end %>
     </div>
+    <% end %>
+
```
Comment (jasononaquest) [inline app/views/users/index.html.erb]: Please remove empty lines between blocks

File: app/assets/javascripts/users.coffee (hunk 1 above)
Comment (justgarcia5) [reply app/assets/javascripts/users.coffee]: I deleted this file.

File: app/assets/stylesheets/users.scss (hunk 2 above)
Comment (justgarcia5) [reply app/assets/stylesheets/users.scss]: I deleted this empty file from app

File: app/assets/stylesheets/scaffolds.scss (hunk 3 above)
Comment (justgarcia5) [reply app/assets/stylesheets/scaffolds.scss]: This file was not being used and was removed

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (justgarcia5) [reply app/controllers/users_controller.rb]: Removed unnecessary comments in the Users Controller

File: app/assets/stylesheets/zeal-initiative-tracker.webflow.css [hunk 18]
```diff
@@ -476,5 +480,92 @@ h5 {
 .icon-target {
   width: 30px;
   height: 30px;
-  margin-right: 13px;
+  margin-right: 13px
```
Comment (jasononaquest) [inline app/assets/stylesheets/zeal-initiative-tracker.webflow.css]: It looks like the semicolon got removed here.

File: app/models/user.rb [hunk 19]
```diff
@@ -0,0 +1,4 @@
+class User < ApplicationRecord
+  validates :fullname, presence: true
```
Comment (jasononaquest) [inline app/models/user.rb:4]: Does Devise already require a password or do we need to validate it here?

File: app/views/users/show.html.erb (hunk 11 above)
Comment (jasononaquest) [reply app/views/users/show.html.erb]: Following up on my previous comment since I realized I might not have been clear: If we can't update users yet, we shouldn't have an edit link at this point.

File: spec/factories.rb [hunk 20]
```diff
@@ -5,4 +5,10 @@
     title 'Test'
     target_date '01/01/2020'
   end
+
+  factory :user do
+    fullname 'Test'
+    email 'email@email.com'
+    # password 'password'
```
Comment (jasononaquest) [inline spec/factories.rb]: If the factory doesn't need a password, go ahead and remove the comment.

File: app/assets/stylesheets/zeal-initiative-tracker.webflow.css (hunk 18 above)
Comment (justgarcia5) [reply app/assets/stylesheets/zeal-initiative-tracker.webflow.css]: I added a semicolon to line 483

File: app/models/user.rb (hunk 19 above)
Comment (justgarcia5) [reply app/models/user.rb:4]: Devise already requires a password

File: spec/factories.rb (hunk 20 above)
Comment (justgarcia5) [reply spec/factories.rb]: This comment was removed

File: app/helpers/users_helper.rb (hunk 5 above)
Comment (justgarcia5) [reply app/helpers/users_helper.rb]: File was deleted

File: app/views/users/_user.json.jbuilder (hunk 6 above)
Comment (justgarcia5) [reply app/views/users/_user.json.jbuilder]: File was deleted

File: app/views/users/index.json.jbuilder (hunk 7 above)
Comment (justgarcia5) [reply app/views/users/index.json.jbuilder]: File was deleted

File: app/views/users/show.json.jbuilder (hunk 8 above)
Comment (justgarcia5) [reply app/views/users/show.json.jbuilder]: file was deleted

File: app/controllers/users_controller.rb (hunk 9 above)
Comment (justgarcia5) [reply app/controllers/users_controller.rb]: Updated this users controller with removing comment and unnecessary methods

File: app/controllers/users_controller.rb (hunk 10 above)
Comment (justgarcia5) [reply app/controllers/users_controller.rb]: This was deleted

File: app/views/users/show.html.erb (hunk 11 above)
Comment (justgarcia5) [reply app/views/users/show.html.erb]: This file was also deleted

File: spec/features/user_spec.rb (hunk 12 above)
Comment (justgarcia5) [reply spec/features/user_spec.rb]: Added more attributes from the story acceptance criteria

File: app/assets/stylesheets/zeal-initiative-tracker.webflow.css (hunk 13 above)
Comment (justgarcia5) [reply app/assets/stylesheets/zeal-initiative-tracker.webflow.css]: This issue was resolved

File: app/assets/stylesheets/zeal-initiative-tracker.webflow.css (hunk 14 above)
Comment (justgarcia5) [reply app/assets/stylesheets/zeal-initiative-tracker.webflow.css]: this issue was resolved

File: app/views/users/edit.html.erb (hunk 15 above)
Comment (justgarcia5) [reply app/views/users/edit.html.erb]: This file was not needed and was removed

File: app/views/users/index.html.erb (hunk 16 above)
Comment (justgarcia5) [reply app/views/users/index.html.erb]: This link was removed. It was not needed

File: app/views/users/index.html.erb (hunk 17 above)
Comment (justgarcia5) [reply app/views/users/index.html.erb]: These empty lines were resolved

File: app/views/initiatives/index.html.erb [hunk 21]
```diff
@@ -25,5 +29,5 @@
         </div>
       <% end %>
     <% end %>
-  </div>
-</div>
+  
+</div>
```
Comment (casto101) [inline app/views/initiatives/index.html.erb:33]: I think you may have lost your closing div tag for your `<div class="init-list">` on line 8 here.

File: app/views/users/_form.html.erb [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+    <%= f.label :is_admin, 'Is Admin', class: 'form-field-label' %>
+    <%= f.check_box :is_admin, {}, "true", "false" %> 
+
+    <%= f.label :team_leader_id, 'Team Leader', class: 'form-field-label' %>
+    <%= f.text_field :team_leader_id, placeholder: 'Enter user id', class: 'form-field w-input' %> 
+    
+    <%= f.submit 'Submit', class: 'btn-primary w-button' %>
+    <% end %>
+
+  </div>
```
Comment (casto101) [inline app/views/users/_form.html.erb:32]: And I think you might have an extra closing div here? Looks like you have one opening tag on line 1 and one on line 2.

File: app/views/users/index.html.erb [hunk 23]
```diff
@@ -0,0 +1,23 @@
+<p id="notice"><%= notice %></p>
+
+<div class="wrapper">
+  <div style="float: right">
```
Comment (casto101) [inline app/views/users/index.html.erb]: What's the thinking behind the inline styling here, as opposed to creating a class for the float behavior and putting it in a style sheet? 

This styling may not actually be applied here, I believe you need a semicolon at the end of the line for the styling to be parsed, ie, `<div style="float:right;">`

**Review by casto101 — CHANGES_REQUESTED** [review]
There are just a couple of things in your HTML that don't quite look right, but otherwise this is really good. Go ahead and look at those and ping me for re-review, and I'll approve.

File: spec/features/user_spec.rb [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+    fill_in 'Full Name', with: 'Test2'
+    fill_in 'Email', with: 'email2@email.com'
+    click_button 'Submit'
+    expect(page).to have_content('Test2')
+    expect(page).to have_content('email2@email.com')
+    expect(page).to have_content('Team Member')
+  end
+end
+
+
```
Comment (casto101) [inline spec/features/user_spec.rb]: Please remove the extra lines at the end of this file, there should only be 1 blank line at the end of the file by default. You might check your editor settings as well; you have a lot of files in this PR that are removing the blank lines at the ends of files, and that's often a "trim whitespace" option that will edit your files by default on save. Our team's practice is to strip whitespace at the ends of line, and add/preserve a single blank line at the end of code files. If not everyone has the same settings on their editor, you could be modifying each others' changes without even realizing it, which adds visual noise to git diffs.

File: app/views/users/_form.html.erb (hunk 22 above)
Comment (justgarcia5) [reply app/views/users/_form.html.erb:32]: Yes there was. I removed the extra div from this file

File: app/views/initiatives/index.html.erb (hunk 21 above)
Comment (justgarcia5) [reply app/views/initiatives/index.html.erb:33]: I added a closing div tag

File: spec/features/user_spec.rb (hunk 24 above)
Comment (justgarcia5) [reply spec/features/user_spec.rb]: Okay. I went through and checked files with extra lines and removed them. I also enabled trim trailing whitespace in vs code.

Comment (casto101) [reply spec/features/user_spec.rb]: The consistency across files is good! But I actually meant that all files should have a single extra blank line at the end of them. I'm sorry, I didn't communicate that very well 😅In VSCode, this setting is called `files.insertFinalNewline`.

File: app/views/users/_form.html.erb (hunk 22 above)
Comment (casto101) [reply app/views/users/_form.html.erb:32]: I don't think your change got saved, I still see 3 closing div tags here and only two opening div tags.

Comment (casto101) [reply app/views/users/_form.html.erb:32]: Whoops sorry, I hadn't refreshed the page. Looks good 👍

**Comment by casto101** [comment]
You will need to merge in master or rebase your changes to reconcile the conflicts in this branch before merging. But this looks good!

**Review by casto101 — APPROVED** [review]

File: app/controllers/users_controller.rb [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+      if @user.save
```
Comment (jasononaquest) [inline app/controllers/users_controller.rb]: Please align this of block with @user above.

**Review by jasononaquest — COMMENTED** [review]
Just a little alignment bit, and resolving the conflicts then 👍

File: app/controllers/users_controller.rb (hunk 25 above)
Comment (kevinyu05) [reply app/controllers/users_controller.rb]: The specified block is aligned

Comment (kevinyu05) [reply app/controllers/users_controller.rb]: The alignment is now corrected
